Fix build with unpatched jpeg 9

zend.h from PHP 5.2 and earlier uses #include <mach-o/dyld.h> which
provides definitions of TRUE and FALSE that conflict with the ones
jpeg 9 will define. So tell jpeg we've already defined them, and what
their type is.

PHP 5.3 and later aren't affected because they no longer #include
<mach-o/dyld.h>; see <https://bugs.php.net/bug.php?id=42630>.
--- a/ext/gd/libgd/gd_jpeg.c.orig	2004-03-29 12:21:00.000000000 -0600
+++ b/ext/gd/libgd/gd_jpeg.c	2017-10-16 20:11:17.000000000 -0500
@@ -39,6 +39,10 @@
 #undef HAVE_STDLIB_H
 
 /* 1.8.1: remove dependency on jinclude.h */
+#ifdef ENUM_DYLD_BOOL
+#define HAVE_BOOLEAN
+typedef enum DYLD_BOOL boolean;
+#endif
 #include "jpeglib.h"
 #include "jerror.h"
 
